### PR TITLE
Refactor/replace heatmap buttons 382

### PR DESCRIPTION
**Pull request by soybrian**
### New MUI Buttons, Custom Control Component

- Add zoom in/out buttons
- Add reset view functionality
- Allow custom positioning on heatmap
- Allow positioning of custom buttons

change1

### CODE REVIEW

File: frontend/src/components/Sidebar/MapComponents/CustomControl.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
+// SPDX-FileCopyrightText: 2024 German Aerospace Center (DLR)
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+
+// Third-party
+import * as am5map from '@amcharts/amcharts5/map';
+import {ZoomIn, ZoomOut, ZoomOutMap} from '@mui/icons-material';
+import {ButtonGroup, IconButton, Stack} from '@mui/material';
```
Comment (JonasGilg) [inline frontend/src/components/Sidebar/MapComponents/CustomControl.tsx:9]: Please import each Type/Class seperately. Please look at the documentation here: https://github.com/DLR-SC/ESID/tree/develop/frontend#imports

File: frontend/src/components/Sidebar/MapComponents/CustomControl.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  const handleReset = () => {
+    if (chart) {
+      try {
+        chart.goHome(300);
+        onHomeClick?.(); // Callback to HeatMap.tsx to synch selected area
+      } catch (error) {
+        console.error('Error while resetting zoom:', error);
+      }
+    }
+  };
```
Comment (JonasGilg) [inline frontend/src/components/Sidebar/MapComponents/CustomControl.tsx:65]: Please wrap the callbacks in a `useCallback` for now.

File: frontend/src/components/Sidebar/MapComponents/HeatMap.tsx [hunk 3]
```diff
@@ -1,21 +1,27 @@
...
+// React imports
+import React, {useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState} from 'react';
+
+// Third-party
 import * as am5 from '@amcharts/amcharts5';
 import * as am5map from '@amcharts/amcharts5/map';
-import {Feature, GeoJSON, GeoJsonProperties} from 'geojson';
-import svgZoomResetURL from '../../../../assets/svg/zoom_out_map_white_24dp.svg?url';
-import svgZoomInURL from '../../../../assets/svg/zoom_in_white_24dp.svg?url';
-import svgZoomOutURL from '../../../../assets/svg/zoom_out_white_24dp.svg?url';
```
Comment (JonasGilg) [inline frontend/src/components/Sidebar/MapComponents/HeatMap.tsx:10]: I think we can delete these assets from the project.

**Review by JonasGilg — CHANGES_REQUESTED** [review]
Great work, looks really good!

I added some comments. also in future please use the PR template.

The CustomControls should be better named, like MapControls or something similar.

File: frontend/src/components/Sidebar/MapComponents/HeatMap.tsx (hunk 3 above)
Comment (soybrian) [reply frontend/src/components/Sidebar/MapComponents/HeatMap.tsx:10]: Deleted.
https://github.com/DLR-SC/ESID/pull/384/commits/59b91ba55b9145fdea75d8cbaa28e3e8bfc240e2

File: frontend/src/components/Sidebar/MapComponents/CustomControl.tsx (hunk 1 above)
Comment (soybrian) [reply frontend/src/components/Sidebar/MapComponents/CustomControl.tsx]: Thanks for the info, fixed.
https://github.com/DLR-SC/ESID/pull/384/commits/9f9a658e46c5e7673927bd331d99dd6d6542bf53

File: frontend/src/components/Sidebar/MapComponents/CustomControl.tsx (hunk 2 above)
Comment (soybrian) [reply frontend/src/components/Sidebar/MapComponents/CustomControl.tsx]: wrapped in useCallback.

https://github.com/DLR-SC/ESID/pull/384/commits/9f9a658e46c5e7673927bd331d99dd6d6542bf53

**Review by JonasGilg — APPROVED** [review]

**Pull request merged**